### PR TITLE
Add new extensions/plugins

### PR DESCRIPTION
**Pull request by TakoB222**
## Issue
The list of supported plugins was too short: https://charmhub.io/postgresql-k8s/docs/r-supported-plugins-extensions

## Solution
Add new extensions/plugins:

bloom
btree_gin
btree_gist
cube
dict_int
dict_xsyn
earthdistance
fuzzystrmatch
intarray
isn
lo
ltree
old_snapshot
pg_freespacemap
pgrowlocks
pgstattuple
pg_visibility
seg
tablefunc
tcn
tsm_system_rows
tsm_system_time
uuid-ossp

### CODE REVIEW

**Review by marceloneppel — APPROVED** [review]
LGTM! Thanks @TakoB222!

**Comment by dragomirp** [comment]
Hi, @TakoB222, can you resync with main? There are some changes there that should fix the check libraries failure.

**Comment by TakoB222** [comment]
> Hi, @TakoB222, can you resync with main? There are some changes there that should fix the check libraries failure.

Yes, sure

File: src/charm.py [hunk 1]
```diff
@@ -555,6 +555,8 @@ def enable_disable_extensions(self, database: str = None) -> None:
             enable = self.config[plugin]
             # Enable or disable the plugin/extension.
             extension = "_".join(plugin.split("_")[1:-1])
+            if extension == "uuid_ossp":
+                extension = '"uuid-ossp"'
```
Comment (dragomirp) [inline src/charm.py:559]: Might be better to add an override mapping so in the future we can add other exceptions there.

File: config.yaml [hunk 2]
```diff
@@ -26,6 +26,98 @@ options:
...
+    type: boolean
+    description: Enable tsm_system_rows extension
+  plugin_tsm_system_time_enable:
+    default: false
+    type: boolean
+    description: Enable tsm_system_time extension
+  plugin_uuid_ossp_enable:
+    default: false
+    type: boolean
+    description: Enable uuid_ossp extension
```
Comment (dragomirp) [inline config.yaml:120]:
```suggestion
    description: Enable uuid-ossp extension
```

File: tests/integration/test_plugins.py [hunk 3]
```diff
@@ -24,6 +24,35 @@
 PG_TRGM_EXTENSION_STATEMENT = "SELECT word_similarity('word', 'two words');"
 PLPYTHON3U_EXTENSION_STATEMENT = 'CREATE FUNCTION plpython_test() RETURNS varchar[] AS $$ return "hello" $$ LANGUAGE plpython3u;'
 UNACCENT_EXTENSION_STATEMENT = "SELECT ts_lexize('unaccent','Hôtel');"
+BLOOM_EXTENSION_STATEMENT = (
```
Comment (dragomirp) [inline tests/integration/test_plugins.py:27]: We can add those in a `plugin`: `SQL test` dictionary and then loop over `.keys()`  in the config definition on [L#192](https://github.com/canonical/postgresql-k8s-operator/blob/bb523a2946c1ddae55e70f219e0da6a1dc0a4dfd/tests/integration/test_plugins.py#L193) and over `.values()` in the checks  on [L#74](https://github.com/canonical/postgresql-k8s-operator/blob/bb523a2946c1ddae55e70f219e0da6a1dc0a4dfd/tests/integration/test_plugins.py#L74) and [L#232](https://github.com/canonical/postgresql-k8s-operator/blob/bb523a2946c1ddae55e70f219e0da6a1dc0a4dfd/tests/integration/test_plugins.py#L232).

**Comment by dragomirp** [comment]
Added a few nitpicks. Best to do them after https://github.com/canonical/postgresql-k8s-operator/pull/281 is merged, since it touches on the same code.

**Comment by TakoB222** [comment]
Okay, thanks for checking, i'll do

**Review by taurus-forever — APPROVED** [review]
LGTM. Thank you for the great contribution here!
Please address Dragomir comments!

**Comment by dragomirp** [comment]
Hi, @TakoB222, https://github.com/canonical/postgresql-k8s-operator/pull/281 was merged. Thanks for the contribution and sorry for all the speed bumps.

**Review by marceloneppel — APPROVED** [review]

**Comment by dragomirp** [comment]
Hi, @TakoB222, please sync main once more. This should sidestep the persistent failures for backup tests on CI.

File: config.yaml [hunk 4]
```diff
@@ -123,6 +123,98 @@ options:
...
+    type: boolean
+    description: Enable tsm_system_rows extension.
+  plugin_tsm_system_time_enable:
+    default: false
+    type: boolean
+    description: Enable tsm_system_time extension.
+  plugin_uuid_ossp_enable:
+    default: false
+    type: boolean
+    description: Enable uuid_ossp extension.
```
Comment (dragomirp) [inline config.yaml:217]:
```suggestion
    description: Enable uuid-ossp extension.
```
We should use the name of the plugin in the description, not the normalised name we have to use for Juju config.

**Comment by TakoB222** [comment]
> Hi, @TakoB222, please sync main once more. This should sidestep the persistent failures for backup tests on CI.

done

**Review by dragomirp — APPROVED** [review]

**Pull request merged**